### ISSUE TITLE
[Enhancement] Replace or clarify concept of 'pinned' keywords

### ISSUE THREAD

**Issue by ator-dev** [searchmarkers/mark-my-search#82]
This issue was created due to a review:
```
What does the pin on the far right mean? If i click on it it disappears but nothing further happens (as far as I can tell).
```

Explanation:
The pin appears if the current keywords were added *without user input*, i.e. automatically. If the pin is visible, the keywords can be overwritten automatically. Automatic keywords normally come from using a search engine.

Problem:
This is unclear and confusing behaviour, especially to new users. The pin does not appear to do anything when clicked, but sets a hidden flag which only occasionally affects behaviour. This should either be better explained and/or presented, or replaced with behaviour that is more consistent and easier to understand.

**Comment by buckleyGI** [comment]
Hi Ator. Thanks for posting this issue.

Its still not clear why the pin is useful? What does it make possible if it is present (or not)?
I both cases, coming from a search engine or not, I can edit the keywords.

One low hanging fruit suggestion for usability is to give the buttons a tooltip

**Comment by ator-dev** [comment]
Here's a description of what happens:

* When you use a search engine, Mark My Search gets invoked automatically and gathers keywords. We cannot assume that the user _wanted_ these specific keywords, but highlighting happens based on the fact that they might.

  * The pin button shows up because the keyword array is given _low priority_.

* If you search when keywords are already present, we have to choose whether or not to overwrite them. If the keyword array has _low priority_ (shows the pin button), Mark My Search will simply discard the old keywords in favour of the new ones.

**Comment by ator-dev** [comment]
> One low hanging fruit suggestion for usability is to give the buttons a tooltip

Agreed, and while I have dynamic 'metrics' tooltips for keyword buttons, I have neglected them for all other controls. I have made a note and will add tooltips in future!

Do you think something like "Do not replace with searched keywords" or "Keep keywords when searching" would make sense?

**Comment by buckleyGI** [comment]
Hmmm this is something that I need to found out when using searchmarkers more. It is clicking a bit but I haven't had a use case where I needed it. "Keep these keywords when searching" sounds good.
Also, the fact that the pin disappears (and is not a toggle) was a bit confusing => What did I just do _and can't undo_

**Comment by ator-dev** [comment]
> I haven't had a use case where I needed it

Thinking more about it, I can see that this might not even be an important feature. After all, most of the time you just want to start from a single search and find everything from there. It adds complexity to the codebase, clutters the UI without a clear reason, and is essentially opaque behaviour that only sometimes has an effect; it's also a lot harder to describe than I realised.

In view of this, I am proposing a different solution. I could make a check that the user has not left the search page before overwriting, but this seems too much like an even more opaque special case. Instead, I could give the user specific control:

* When a search is detected which produces _different keywords from the current ones_, a 'refresh' button is shown in the toolbar (where the pin button would be).

* If you ignore it, nothing happens and it will disappear when you navigate somewhere else; if you press it, the keywords are immediately replaced by the ones detected in your search.

* A keyboard shortcut would be made available for this action.

**Comment by ator-dev** [comment]
Implemented for __v1.12.0__. This "auto overwriting" feature has been baked into Mark My Search since v1.0 - with code being built for and around it since then - so it has taken significant refactoring to remove it. I have maintained a high standard for code quality so the process has been relatively painless, so with testing there should be no additional bugs on release. I took the opportunity to improve the surrounding code in what is already a simplification, so this is definitely a net positive.

![image](https://user-images.githubusercontent.com/66553618/212997268-c7fe8867-ae0b-4f55-89b5-55503d8c3b96.png)

I think the icon is acceptable, but it will definitely be a candidate for replacement (along with the 'eye') when I find time for an icon overhaul.
